plugins/digitalocean/test/test_collector.py

```diff
@@ -0,0 +1,345 @@
+    check_edges(graph, "do:team:test_team", "do:region:fra1")
+    check_edges(graph, "do:region:fra1", "do:droplet:289110074")
+
+    # region nyc1 should not be in the graph since it has no resources in it
+    assert graph.search_first("id", "do:region:nyc1") is None
+
+    region = graph.search_first("id", "do:region:fra1")
+    assert region.name == "Frankfurt 1"
+    assert region.kind == "digitalocean_region"
+    assert region.slug == "fra1"
+    assert region.features == [
+        "backups",
+        "ipv6",
+        "metadata",
+        "install_agent",
+        "storage",
+        "image_transfer",
+    ]
+    assert region.available is True
+
+
+def test_collect_vpcs():
+    do_client = ClientMock(
+        {
+            "list_regions": regions,
+            "list_vpcs": vpcs,
+        }
+    )
+    graph = prepare_graph(do_client)
+
+    check_edges(graph, "do:region:fra1", "do:vpc:0d3176ad-41e0-4021-b831-0c5c45c60959")
+    vpc = graph.search_first("id", "do:vpc:0d3176ad-41e0-4021-b831-0c5c45c60959")
+    assert vpc.id == "do:vpc:0d3176ad-41e0-4021-b831-0c5c45c60959"
+    assert vpc.name == "default-fra1"
+    assert vpc.description == ""
+    assert vpc.ip_range == "127.0.0.1/20"
+    assert vpc.default is True
+
+
+def test_collect_droplets():
+    do_client = ClientMock(
+        {
+            "list_regions": regions,
+            "list_droplets": droplets,
+            "list_vpcs": vpcs,
+        }
+    )
+    graph = prepare_graph(do_client)
+
+    check_edges(graph, "do:region:fra1", "do:droplet:289110074")
+    check_edges(
+        graph, "do:vpc:0d3176ad-41e0-4021-b831-0c5c45c60959", "do:droplet:289110074"
+    )
+    droplet = graph.search_first("id", "do:droplet:289110074")
+    assert droplet.id == "do:droplet:289110074"
+    assert droplet.name == "ubuntu-s-1vcpu-1gb-fra1-01"
+    assert droplet.instance_memory == 1024
+    assert droplet.instance_cores == 1
+    assert droplet.instance_status == "running"
+    assert droplet.region().id == "do:region:fra1"
+    assert droplet.image == "ubuntu-20-04-x64"
+    assert droplet.locked is False
+    assert droplet.ctime == datetime.datetime(
+        2022, 3, 3, 16, 26, 55, tzinfo=datetime.timezone.utc
+    )
+    assert droplet.tags == {}
+
+
+def test_collect_volumes():
+    do_client = ClientMock(
+        {
+            "list_regions": regions,
+            "list_droplets": droplets,
+            "list_volumes": volumes,
+        }
+    )
+    graph = prepare_graph(do_client)
+
+    check_edges(
+        graph, "do:droplet:289110074", "do:volume:631f81d2-9fc1-11ec-800c-0a58ac14d197"
+    )
+    volume = graph.search_first("id", "do:volume:631f81d2-9fc1-11ec-800c-0a58ac14d197")
+    assert volume.id == "do:volume:631f81d2-9fc1-11ec-800c-0a58ac14d197"
+    assert volume.name == "volume-fra1-01"
+    assert volume.description == "Test volume"
+    assert volume.filesystem_type == "ext4"
+    assert volume.filesystem_label == ""
```

Comment:
because it has no label? or the label is really an empty string?

Reply:
fixed